src/utils/ipv6.hpp

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <cstring>
+#include <iterator>
+#include <arpa/inet.h>
+
+#include "ipv6.hpp"
+
+namespace
+{
+    constexpr char ipv4_mapped_ipv6_prefix[] = "::ffff:";
+    constexpr std::size_t ipv4_mapped_ipv6_prefix_length =
+        std::size(ipv4_mapped_ipv6_prefix);
+}
+
+[[nodiscard]]
+const char *get_ipv4_mapped_ipv6_from_ipv6(const char *ipv6_address) noexcept
+{
+    return !std::strncmp(ipv6_address,
+                         ipv4_mapped_ipv6_prefix,
+                         ipv4_mapped_ipv6_prefix_length) ?
+        ipv6_address + ipv4_mapped_ipv6_prefix_length : nullptr;
+}
+
+[[nodiscard]]
+bool is_ipv6_address(const char *ipv6_address) noexcept
+{
+    sockaddr_in6 s6;
+    
+    return ::inet_pton(AF_INET6, ipv6_address, &s6.sin6_addr) == 1;
+}
+
+[[nodiscard]]
+bool convert_ipv4_mapped_ipv6_to_ipv6(const char *ipv4_address,
+                                      char *ipv6_address,
```

Comment:
`char *ipv6_address, std::size_t ipv6_address_length` -> `writable_bytes_view ipv6_address` or `array_view`.

Reply:
So, if you suggest array_view, is it good to use also std::string_view?

Because I see that in your array_view.hpp implementation, the use of array_view<char> will be like std::string_view